playground/backend/internal/fs_tool/compiled_tool.go

```diff
@@ -0,0 +1,47 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_tool
+
+import (
+	"github.com/google/uuid"
+	"path/filepath"
+)
+
+const (
+	baseFileFolder     = "executable_files"
+	sourceFolderName   = "src"
+	compiledFolderName = "bin"
+)
+
+// newCompiledLifeCycle creates LifeCycle with compiled SDK environment.
+func newCompiledLifeCycle(pipelineId uuid.UUID, workingDir string, executableFileExtension string, compiledFileExtension string) *LifeCycle {
```

Comment:
compiled life cycle also doesn't sound good
javaLifeCycle, goLifeCycle sounds fine, compiled is not clear what it is